src/main/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGenerator.java

```diff
@@ -332,4 +332,61 @@ private boolean isMatchingByLineAndColumnAndTokenType(DetailAST ast) {
                 && expandedTabColumn(ast) == columnNumber
                 && (tokenType == 0 || tokenType == ast.getType());
     }
+
+    /**
+     * Controls true escape characters. True escape characters need '&amp;' before but it also
+     * requires XML 1.1 until https://github.com/checkstyle/checkstyle/issues/5168.
+     *
+     * @param chr character to check.
+     * @param encStr StringBuilder from parent method encode().
+     */
+    private static void controlTrueCharacter(char chr, StringBuilder encStr) {
```

Comment:
please move this method to be after "encode" method, we care about order of method declarations.

Reply:
name of this method is weird.
What is "true escape character" ?

Can you inline method in switch ? to make sure we see all `append` in same place

Reply:
Even I don't know but I didn't change it since it was mentioned in [XMLLogger](https://github.com/checkstyle/checkstyle/blob/5f53bfe41e96a0fcd0a8266ffc0db0fb32c54d6e/src/main/java/com/puppycrawl/tools/checkstyle/XMLLogger.java#L275) for the exact same thing, so I didn't change the description. I had changed it now.

Reply:
I didn't get by **inline method in switch**. Can you please elaborate a bit.

Reply:
Move whole body of this method to switch code block. Remove this method at all

Reply:
Sir, the issue with it is the `encode` method is becoming larger than 30 lines, and then it gives violation.

Reply:
please read https://github.com/checkstyle/checkstyle/pull/9247#discussion_r576457648